lib/globus/client/endpoint.rb

```diff
@@ -22,22 +22,18 @@ def length
 
       # Create a directory https://docs.globus.org/api/transfer/file_operations/#make_directory
       def mkdir
-        path = config.uploads_directory
```

Comment:
Instead of mutating a string variable to build up a directory tree, I made a change in this PR to instead use iteration.